luarules/gadgets/game_team_power_watcher.lua

```diff
@@ -62,233 +29,462 @@ if gadgetHandler:IsSyncedCode() then
         {threshold = 475000, techLevel = 3.5},
         {threshold = 600000, techLevel = 4},
         {threshold = 725000, techLevel = 4.5}
-    }
+    },
+
+
+    --stores the values of powers for teams
+    PeakTeamPowers = {},
+    TeamPowers = {},
+
+    --stores the power of each living unit as a table {unitID, power, unitTeam}
+    UnitsWithPower = {},
+
+    
+    --teams
+    ScavengerTeam = nil,
+    RaptorTeam = nil,
+    NeutralTeam = nil,
+    AITeams = {},
+    HumanTeams = {}
+}
+
+local spGetGameSeconds = Spring.GetGameSeconds
+
+local teamList = Spring.GetTeamList()
 
+local testPowerTable = {}
+local testPowerNumber = 0
 
 --decipher human vs ai vs neutral vs defense mode ai's 
-    for _, teamID in ipairs(teamList) do
-		local teamLuaAI = Spring.GetTeamLuaAI(teamID)
-		if (teamLuaAI and string.find(teamLuaAI, "ScavengersAI")) then
-			scavTeam = teamID
-        elseif (teamLuaAI and string.find(teamLuaAI, "RaptorsAI")) then
-            raptorTeam = teamID
-        elseif select (4, Spring.GetTeamInfo(teamID, false)) then
-            AIteams[teamID] = true
-        elseif teamID == tonumber(teamList[#teamList]) then
-            neutralTeamNumber = teamID
-        else
-			humanTeams[teamID] = true
-		end
-	end
+for _, teamID in ipairs(teamList) do
+    local teamLuaAI = Spring.GetTeamLuaAI(teamID)
+    if (teamLuaAI and string.find(teamLuaAI, "ScavengersAI")) then
+        GG.PowerLib.ScavengerTeam = teamID
+    elseif (teamLuaAI and string.find(teamLuaAI, "RaptorsAI")) then
+        GG.PowerLib.RaptorTeam = teamID
+    elseif select (4, Spring.GetTeamInfo(teamID, false)) then
+        GG.PowerLib.AITeams[teamID] = true
+    elseif teamID == tonumber(teamList[#teamList]) then
```

Comment:
Is the neutral (Gaia) team guaranteed to be numerically last team ID?

Reply:
Yes, it's done this way across all the PVE gadgets I've seen. Never seen an exception. Thanks for checking.